src/Config/ObjectModel/RuntimeOptions.cs

```diff
@@ -3,4 +3,9 @@
 
 namespace Azure.DataApiBuilder.Config.ObjectModel;
 
-public record RuntimeOptions(RestRuntimeOptions Rest, GraphQLRuntimeOptions GraphQL, HostOptions Host);
+public record RuntimeOptions(RestRuntimeOptions Rest, GraphQLRuntimeOptions GraphQL, HostOptions Host, string? BaseRoute = null)
+{
+    public const string JSON_PROPERTY_NAME = "runtime";
+    public const string PROPERTY_NAME_PATH = "path";
+    public const string PROPERTY_NAME_BASE_ROUTE = "base-route";
```

Comment:
I have reservations about this as it means we're starting to bleed out the structure of the JSON config to other parts of the codebase.

We shouldn't need to say "this is what the JSON looks like", that should only be the concern of the serializer/deserializer.

I see where this is being used, downstream in the validator to push into the potential exception messages. This is a risk factor created by really generic methods (such as those in the validator), and I wonder if we can't achieve the same with a `nameof` call against the relevant property, especailly as `runtime` and `path` aren't properties of `RuntimeOptions` so we're kind of shoving stuff in places that they don't directly link to.

Reply:
I see what you're saying. @aniruddh25 also had a similar comment on the other PR: https://github.com/Azure/data-api-builder/pull/1496#discussion_r1253854635. You can have a look there.

Reply:
I posted a comment elsewhere with suggestions on how to avoid adding these string properties.

Reply:
I have made the possible reduction in passing of these constants as arguments but if we totally want to remove these constants, we need to use string literals.

Reply:
I'm still on the fence about whether explicit constants are necessary. Was the motivation to have fewer places to change a value in case we ever change the JSON property name? As any changes to property names would be breaking changes, that scenario seems unlikely.

Reply:
I'm with @seantleonard here, I feel like it's tackling a hypothetical scenario that, if it came to be, would have more cascading problems than mistakenly changing a string value.

Reply:
Yeah, agree with you two on this. Thanks, incorporated this.